packages/bsp/common/etc/update-motd.d/30-armbian-sysinfo

```diff
@@ -156,7 +156,7 @@ display "Battery" "$battery_percent" "20" "1" "%" "$status_battery_text" ; a=$((
 (( $a > 0 )) && echo "" # new line only if some value is displayed
 
 line=0
-if [[ "$(command -v vnstat)" && $(vnstat --iflist | grep -c "$PRIMARY_INTERFACE") -eq 1 ]]; then
+if [[ "$(command -v vnstat)" && $(vnstat --iflist | grep -c "$PRIMARY_INTERFACE") -eq 1 && $(vnstat -i $PRIMARY_INTERFACE --oneline | cut -d";" -f4,5) != *Error* ]]; then
```

Comment:
The `cut` command is not necessary here, and in fact this entire line can be simplified to just:
```
if vnstat -i "$PRIMARY_INTERFACE" &> /dev/null; then
```

Reply:
OK, but it needs at least one more check - it not empty.

Reply:
No, it doesn't matter. If `$PRIMARY_INTERFACE` is empty, then the command is just `vnstat -i`, which will fail, same as if it contained an invalid interface.

Reply:
It won't fail:

```
root@odroidxu4:/etc/default# echo $PRIMARY_INTERFACE

root@odroidxu4:/etc/default# vnstat -i "$PRIMARY_INTERFACE"
Database updated: 2022-12-24 21:10:00

   enx001e0636e380 since 2022-08-15

          rx:  2.37 GiB      tx:  41.65 GiB      total:  44.02 GiB

   monthly
                     rx      |     tx      |    total    |   avg. rate
     ------------------------+-------------+-------------+---------------
       2022-11    383.54 MiB |    1.14 GiB |    1.52 GiB |    5.02 kbit/s
       2022-12    374.29 MiB |    5.54 GiB |    5.91 GiB |   24.58 kbit/s
     ------------------------+-------------+-------------+---------------
     estimated    485.32 MiB |    7.19 GiB |    7.66 GiB |

   daily
                     rx      |     tx      |    total    |   avg. rate
     ------------------------+-------------+-------------+---------------
     yesterday     50.23 MiB |  553.22 MiB |  603.45 MiB |   58.59 kbit/s
         today     35.36 MiB |    2.08 GiB |    2.11 GiB |  238.17 kbit/s
     ------------------------+-------------+-------------+---------------
     estimated     40.09 MiB |    2.36 GiB |    2.40 GiB |

```

Reply:
This does fail:
vnstat -i $PRIMARY_INTERFACE